iceoryx_dust/include/iceoryx_dust/cxx/std_chrono_support.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_STD_CHRONO_SUPPORT_HPP
+#define IOX_DUST_STD_CHRONO_SUPPORT_HPP
+
+#include "iox/duration.hpp"
+
+#include <chrono>
+
+namespace iox
+{
+namespace units
+{
+/// @brief Construct a Duration object from std::chrono::milliseconds
+/// @param[in] value as milliseconds
+/// @attention Since negative durations are not allowed, the duration will be clamped to 0
+Duration make_duration(const std::chrono::milliseconds value) noexcept;
+
+/// @brief Construct a Duration object from std::chrono::nanoseconds
+/// @param[in] value as nanoseconds
+/// @attention Since negative durations are not allowed, the duration will be clamped to 0
+Duration make_duration(const std::chrono::nanoseconds value) noexcept;
```

Comment:
Why not just `into<Duration>`? It is similar to the `std::string -> iox::string` approach and when we use it for other type conversions a user only needs to learn the pattern once. The same can be done for other STL types if necessary.

In addition we could also have a `into<std::chrono::milliseconds>` later on.

Reply:
Good point, I'll refactor that :sweat_smile: